### PR TITLE
Use ArraySet.

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
`ArraySet` is a more memory-efficient implementation of `Set` than `HashSet`, and it supports index-based value retrieval.

### CODE REVIEW

**Review by cooltey — APPROVED** [review]

File: app/build.gradle [hunk 1]
```diff
@@ -196,6 +196,7 @@ dependencies {
 
     implementation "com.google.android.material:material:1.5.0"
     implementation 'androidx.appcompat:appcompat:1.4.1'
+    implementation "androidx.collection:collection-ktx:1.2.0"
```
Comment (dbrant) [inline app/build.gradle:199]: It doesn't seem like we need to declare this dependency explicitly here.

File: app/src/main/java/org/wikipedia/page/PageActivity.kt [hunk 2]
```diff
@@ -391,9 +393,9 @@ class PageActivity : BaseActivity(), PageFragment.Callback, LinkPreviewDialog.Ca
     }
 
     override fun onPageCloseActionMode() {
-        val actionModesToFinish = HashSet(currentActionModes)
-        for (mode in actionModesToFinish) {
-            mode.finish()
+        val actionModesToFinish = ArraySet(currentActionModes)
+        for (i in actionModesToFinish.indices) {
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/page/PageActivity.kt:397]: Why not keep the previous (cleaner) syntax of `for (mode in actionModesToFinish)`?

**Review by dbrant — CHANGES_REQUESTED** [review]

File: app/build.gradle (hunk 1 above)
Comment (Isira-Seneviratne) [reply app/build.gradle:199]: In the older version, the return type of `valueAt` was marked as nullable.

File: app/src/main/java/org/wikipedia/page/PageActivity.kt (hunk 2 above)
Comment (Isira-Seneviratne) [reply app/src/main/java/org/wikipedia/page/PageActivity.kt:397]: The documentation [recommends](https://developer.android.com/reference/androidx/collection/ArraySet#iterator()) against using the iterator.

**Comment by dbrant** [comment]
Same comments/findings as #3290.

**Pull request closed without merging**